### PR TITLE
TC1, TC2 for capacitor

### PR DESCRIPTION
**Pull request by marcin-golebiowski**

### CODE REVIEW

File: SpiceSharp/Components/RLC/CAP/TemperatureBehavior.cs [hunk 1]
```diff
@@ -98,15 +106,25 @@ public void Temperature(BaseSimulation simulation)
...
                               (BaseParameters.Length - ModelParameters.Narrow) +
                               ModelParameters.JunctionCapSidewall * 2 * (
                                   BaseParameters.Length - ModelParameters.Narrow +
                                   (width - ModelParameters.Narrow));
             }
             else
-                Capacitance = BaseParameters.Capacitance;
+                capacitance = BaseParameters.Capacitance;
+
+            double difference = BaseParameters.Temperature - (ModelParameters?.NominalTemperature.Value ?? Circuit.ReferenceTemperature);
```
Comment (svenboulanger) [inline SpiceSharp/Components/RLC/CAP/TemperatureBehavior.cs]: This line is only necessary if `ModelParameters != null`, should be moved inside the check.

File: SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs [hunk 2]
```diff
@@ -65,6 +66,95 @@ public void When_Capacitor_LowpassRCTransient_Expect_Reference()
             AnalyzeTransient(tran, ckt, exports, references);
         }
 
+        [Test]
+        public void When_Capacitor_Has_TC1_Zero_TC2_Zero_And_30Degrees_LowpassRCTransient_Expect_Reference()
```
Comment (svenboulanger) [inline SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs]: This method name is too long. Use the same format as the others test methods.

File: SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs [hunk 3]
```diff
@@ -65,6 +66,95 @@ public void When_Capacitor_LowpassRCTransient_Expect_Reference()
...
+
+            Export<double>[] exports = { new RealPropertyExport(tran, "C1", "v") };
+            Func<double, double>[] references = { t => dcVoltage * (1.0 - Math.Exp(-t / tau)) };
+
+            // Run
+            AnalyzeTransient(tran, ckt, exports, references);
+        }
+
+        [Test]
+        public void When_Capacitor_Has_TC1_NonZero_NonTC2_Zero_And_30Degrees_LowpassRCTransient_Expect_Reference()
```
Comment (svenboulanger) [inline SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs]: Same naming issue.

File: SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs [hunk 4]
```diff
@@ -65,6 +66,95 @@ public void When_Capacitor_LowpassRCTransient_Expect_Reference()
...
+                new VoltageSource("V1", "IN", "0", dcVoltage)
+            );
+
+            // Create simulation, exports and references
+            var tran = new Transient("tran", 1e-8, 10e-6);
+            tran.Configurations.Get<TimeConfiguration>().InitialConditions["OUT"] = 0.0;
+
+            tran.BeforeTemperature += (sender, args) =>
+                {
+                    ((BaseSimulationState)args.State).Temperature = Circuit.CelsiusKelvin + 30.0;
```
Comment (svenboulanger) [inline SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs]: This is not the envisioned way to specify a circuit temperature. `BaseConfiguration`. is missing this parameter so this is a missing feature... Needs to be fixed (perhaps later).

Comment (marcin-golebiowski) [reply SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs]: @svenboulanger Good to know. I use this hack in a few other important places in the parser.

File: SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs (hunk 3 above)
Comment (marcin-golebiowski) [reply SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs]: @svenboulanger Changed.

File: SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs (hunk 2 above)
Comment (marcin-golebiowski) [reply SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs]: @svenboulanger Changed

Comment (svenboulanger) [reply SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs]: This project uses convention number 6 from [here](https://dzone.com/articles/7-popular-unit-test-naming). The new names are still not following this convention (the underschore after `Capacitor` should be removed.

Comment (marcin-golebiowski) [reply SpiceSharpTest/Models/RLC/CAP/CapacitorTests.cs]: @svenboulanger Ok. I've changed name again.

**Pull request merged**